### PR TITLE
Upgrade tests on CI

### PR DESCRIPTION
**Pull request by tsusanka**
Only for legacy, I'll patch core in the meantime.

- @matejcik could you review the python part? The test script and wrapper are written by @prusnak 
- Should I create a make target?
- The tests should be probably in tests/ root directory, I've created an issue for that #208 
- The Travis tests are failing, because it can't find the artificats repo (which is on internal GitLab). There might be a [fix](https://stackoverflow.com/questions/52179463/how-to-exclude-a-specific-git-submodule-from-update), but I'm not sure we'll be able to instruct Travis like that, since the submodules init is done automatically on their side. I would like to change this anyway and have the binaries on a "testers" server. We can discuss that in person.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -22,3 +22,6 @@
 	path = vendor/nanopb
 	url = https://github.com/nanopb/nanopb.git
 	ignore = untracked
+[submodule "vendor/artifacts"]
```
Comment (prusnak) [inline .gitmodules]: I don't like this being included as a submodule. It makes impossible for outsiders to properly check-out the project. We might want to become this part of the internal Dockerfile. Let's discuss this IRL.

File: python/trezorlib/client.py [hunk 2]
```diff
@@ -112,6 +112,7 @@ def __init__(self, transport, ui=_NO_UI_OBJECT, state=None):
         self.transport = transport
         self.ui = ui
         self.state = state
+        self.check_firmware = True
```
Comment (matejcik) [inline python/trezorlib/client.py]: let's not make it _this_ easy for 3rd parties to skip the version check.

we might do what Electrum does (`TrezorClient.is_outdated = lambda self: True`)

or better, we can override `trezorlib.MINIMUM_FIRMWARE_VERSION` value in the test runner

Comment (tsusanka) [reply python/trezorlib/client.py]: Yes, I agree. I've added this option to set this to False in debug client only, but I understand 3rd parties could still do so.

> or better, we can override trezorlib.MINIMUM_FIRMWARE_VERSION value in the test runner

I like that. Will do

**Comment by tsusanka** [comment]
I'll file a new one to make this clearer.

**Pull request closed without merging**